addons/account/models/account_move.py

```diff
@@ -4935,6 +4944,41 @@ def _post(self, soft=True):
         # reconcile if state is in draft and move has reversal_entry_id set
         draft_reverse_moves = to_post.filtered(lambda move: move.reversed_entry_id and move.reversed_entry_id.state == 'posted')
 
+        # deal with the eventually related draft moves to the ones we want to post
+        partials_to_unlink = self.env['account.partial.reconcile']
+
+        for move in self:
+            partial_vals, dummy = move._get_reconciled_invoices_partials()
+            for (partial, dummy, counterpart_move_line) in partial_vals:
+                counterpart_move = counterpart_move_line.move_id
+
+        for aml in self.line_ids:
+            for partials, counterpart_field in [(aml.matched_debit_ids, 'debit_move_id'), (aml.matched_credit_ids, 'credit_move_id')]:
+                for partial in partials:
+                    counterpart_move =  partial[counterpart_field].move_id
+                    if counterpart_move.state == 'posted' or counterpart_move in to_post:
+                        if partial._get_draft_exchange_move_vals() != partial.draft_exchange_move_vals:
```

Comment:
I have the feeling a `if partial.exchange_move_id` is missing here because `draft_exchange_move_vals` won't be set on a partial not linked to an exchange diff, right?

Reply:
no actually, there's one case where we can't rely on exchange_move_id being filled or not: if the draft reconciliation didn't create a draft extra move at that time, but now that we're posting it we realize some values changed and we actually should be creating an extra move. 

For example, I encode a draft invoice on 1st january in foreign currency, encode the payment the 1st January as well and post it. Now back on the draft invoice, I switch the invoice date to something else where a currency diff would be needed.